src/client.ts

```diff
@@ -79,15 +80,21 @@ export class Client {
     clientConfiguration?: Partial<ClientConfiguration>,
     httpClient?: HTTPClient
   ) {
-    this.#clientConfiguration = {
-      ...defaultClientConfiguration,
-      ...clientConfiguration,
-      secret: this.#getSecret(clientConfiguration),
-    };
+    this.#clientConfiguration = applyDefaults(
```

Comment:
There is no way to be selective about which fields are permitted to be explicitly provided as undefined or not. JS is just crappy that way. So, since we know that the driver will behave incorrectly when some options are undefined, I can instead enumerate each one and individually ignore undefined values. For client configuration, at this time, that will amount to the exact same behavior (except for max_conns, because that's not connected to anything yet).

Reply:
yeah i understand that, but we could expand validateCOnfiguration to catch those cases and keep the interface very simple to understand.

Treating undefined special makes the interface more complex

Reply:
Updated to throw error when the required options are undefined. We will no longer apply the defaults in that case.